### PR TITLE
Showing the Java environment used by Dataflow in the UI

### PR DESCRIPTION
**Pull request by pabloem**

### CODE REVIEW

**Comment by pabloem** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by pabloem** [comment]
I need to check how these look in the UI

**Comment by pabloem** [comment]
This is showing up in the user agent - probably not what we want:
image

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java [hunk 1]
```diff
@@ -326,8 +326,15 @@ public static DataflowRunner fromOptions(PipelineOptions options) {
     }
 
     DataflowRunnerInfo dataflowRunnerInfo = DataflowRunnerInfo.getDataflowRunnerInfo();
+    // Adding the Java version to the SDK name for user's and support convenience.
+    String javaVersion =
+        Float.parseFloat(System.getProperty("java.specification.version")) >= 9
+            ? "(JDK 11 environment)"
+            : "(JRE 8 environment)";
     String userAgent =
```
Comment (Ardagan) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java]: Can you elaborate why we mix jre and jdk here?

My understanding of this article is that java.specification.version adheres to current jre specification.
https://docs.oracle.com/javase/7/docs/technotes/guides/versioning/spec/versioning2.html

Comment (pabloem) [reply runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java]: Yeah. For Java 11 there is no "JRE" - the only distribution for Java is the JDK. The containers that we provide have the Java 11 JDK.

For Java 8, there is a development distribution, and a runtime distribution (JRE). For the containers we provide the JRE.

Comment (Ardagan) [reply runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java]: Thank you for clarification.

**Comment by tvalentyn** [comment]
What's next on this PR?

**Comment by pabloem** [comment]
Run Dataflow ValidatesRunner Java 11

**Comment by pabloem** [comment]
This looks like this:
image
image

r: @aaltay

**Comment by pabloem** [comment]
Run JavaPortabilityApi PreCommit

**Comment by pabloem** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by pabloem** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by pabloem** [comment]
Run Dataflow ValidatesRunner Java 11

**Comment by pabloem** [comment]
Run Dataflow ValidatesRunner Java 11

**Comment by pabloem** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by pabloem** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by pabloem** [comment]
Run Dataflow ValidatesRunner Java 11

**Comment by pabloem** [comment]
I guess this change affects the SDK version matching, so we can't use it.

**Comment by pabloem** [comment]
This is what it will look like. Is this reasonable?
r: @aaltay 

> image

**Pull request merged**